### PR TITLE
Protects dorms from radstorms

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request

Title.

This method is useful because it prevents dorms itself from being radproof, only the rooms.

## Why It's Good For The Game

erp 

## Proof Of Testing

i havent actually tested this

## Changelog

:cl:
fix: Dorm rooms are now radstorm-proof
/:cl:

### CODE REVIEW

File: modular_zubbers/code/modules/radstorm_proof_dorms/radiation_storm.dm [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/datum/weather/rad_storm/New()
+	. = ..()
+
+	for (var/area/station/commons/dorms/dorm_subtype in subtypesof(/area/station/commons/dorms))
+		protected_areas += dorm_subtype
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/radstorm_proof_dorms/radiation_storm.dm:5]: Move this to `modular_zubbers/code/datums/weather/weather_types/radiation_storm.dm`
Bonus points for also moving the code in `modular_zubbers/maps/offstation/dauntless/areas.dm#L318` there

**Review by Majkl-J — CHANGES_REQUESTED** [review]

File: modular_zubbers/code/datums/weather/weather_types/radiation_storm.dm [hunk 2]
```diff
@@ -9,3 +9,13 @@
...
+
+
+/datum/weather/rad_storm/New()
+	. = ..()
+
+	for (var/area/station/commons/dorms/dorm_subtype in subtypesof(/area/station/commons/dorms))
+		protected_areas += dorm_subtype
+
+	protected_areas += /area/ruin/space/has_grav/bubbers/dauntless
+	protected_areas += /area/ruin/space/has_grav/bubbers/dauntless_space
```
Comment (Majkl-J) [inline modular_zubbers/code/datums/weather/weather_types/radiation_storm.dm:21]: Pile these into the lazyor above, also it already includes subtypes so no need for the loop

**Review by Majkl-J — CHANGES_REQUESTED** [review]

File: modular_zubbers/code/datums/weather/weather_types/radiation_storm.dm (hunk 2 above)
Comment (nikothedude) [reply modular_zubbers/code/datums/weather/weather_types/radiation_storm.dm:21]: commons/dorms protects ALL of dorms, im aiming to protect only rooms

if youre fine with protecting all of dorms, i can make it simpler but yeah

Comment (Majkl-J) [reply modular_zubbers/code/datums/weather/weather_types/radiation_storm.dm:21]: I feel like just having all of dorms protected

**Review by StrangeWeirdKitten — APPROVED** [review]

**Comment by lessthnthree** [comment]
https://github.com/Skyrat-SS13/Skyrat-tg/pull/29682

**Comment by StrangeWeirdKitten** [comment]
Seems to be open upstream

**Pull request closed without merging**